### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by GuiltyDolphin**
Using the new GitHub system, we can have Pull Request Templates to ensure the information we need is present in pull requests!

@duckduckgo/community-leaders What information do *you* want in here?

/cc @moollaza @zachthompson @talsraviv @duckduckgo/community-leaders

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
@GuiltyDolphin let's make it an .md file and put it in a .github folder

**Comment by GuiltyDolphin** [comment]
@moollaza Done. Wasn't 100% sure it accepted `.md` (though I don't see why it wouldn't!)

**Comment by zachthompson** [comment]
I'd like to see "enhancements" and "bug fixes" distinguished since they have different priorities.  Could probably use a section for non-IA related changes as well.

**Comment by GuiltyDolphin** [comment]
@zachthompson Is that good? If you want more (or less) sections then just let me know!

**Comment by gautamkrishnar** [comment]
LGTM :+1:

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+Choose the most relevant items and use the following title template to name
+your Pull Request.
+
+- [ ] New Instant Answer
+    - [ ] Cheat Sheets: **`New {Cheat Sheet Name} Cheat Sheet`**
+    - [ ] Other: **`New {IA Name} Instant Answer`**
+- [ ] Improvement
+    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
+    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
+- [ ] Non Instant Answer
```
Comment (zachthompson) [inline .github/PULL_REQUEST_TEMPLATE.md]: Hyphen

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    - [ ] Role: Use same format as other sections, but append `GoodieRole` to the name.
+    - [ ] Documentation: **`Docs: {Description of Improvements}`**
+    - [ ] File (e.g., `CONTRIBUTING.md` or `.gitignore`): **`{File Name}: {Description of Improvements}`**
```
Comment (zachthompson) [inline .github/PULL_REQUEST_TEMPLATE.md]: What if there are multiple files?

**Comment by zachthompson** [comment]
@GuiltyDolphin looks pretty good overall. @moollaza will have a better idea of any other info that would be helpful.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (GuiltyDolphin) [inline .github/PULL_REQUEST_TEMPLATE.md]: @zachthompson But where?

`Non-Instant Answer` would mean an Answer that wasn't Instant.

`Non-Instant-Answer` is getting a bit off...

I think a different phrasing is best. Any suggestions? `Other Changes`? `Changes not related to Instant Answers`?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (GuiltyDolphin) [inline .github/PULL_REQUEST_TEMPLATE.md]: @zachthompson It is intended just for single-file changes of the main files, e.g., `CONTRIBUTING.md: Correct typo`.

Comment (zachthompson) [inline .github/PULL_REQUEST_TEMPLATE.md]: I don't know.  Something seems inconsistent about it.  Why not do the same with cheat sheets, bug fix, documentation, etc.?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (zachthompson) [inline .github/PULL_REQUEST_TEMPLATE.md]: Ok, then an [en dash](http://english.stackexchange.com/questions/78050/using-non-to-prefix-a-two-word-phrase), e.g. Non&ndash;Instant Answer, which is the recommendation or rephrase.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (GuiltyDolphin) [inline .github/PULL_REQUEST_TEMPLATE.md]: @zachthompson Hmm? I'm not sure I follow - do you mean to have cheat sheets as `vim.json: Fix blah`? I think `Vim Cheat Sheet: Fix Blah` is easier to glance at and quickly get an understanding of the PR.

I can remove this category if you wish - I don't think we get PRs modifying the top-level files very often.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (GuiltyDolphin) [inline .github/PULL_REQUEST_TEMPLATE.md]: @zachthompson Oh, haha, hadn't come across that before — awesome.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (zachthompson) [inline .github/PULL_REQUEST_TEMPLATE.md]: I'm just saying you have this category for a single file when we really need something more general, e.g. Templates, Test Overhaul, etc.  Also, we probably still would like to know if this change is a bug fix, enhancement, etc.

Comment (GuiltyDolphin) [inline .github/PULL_REQUEST_TEMPLATE.md]: @zachthompson Ah!

I think as we get more *obscure* it *should* become less of an issue (usually people fiddling with those will use decent naming); but yes - we should standardize!

Not 100% sure how to do this in general though - without listing everything; what would we call these... 'groups' of related files?

Comment (zachthompson) [inline .github/PULL_REQUEST_TEMPLATE.md]: It's the same as Role, almost like a subject or sub-area.  There aren't too many of them so it wouldn't be crazy to list a few examples and combine with `Role` maybe:
```
Role, Template, Test, Docs: short description
```
These should probably all have the same format.

Comment (GuiltyDolphin) [inline .github/PULL_REQUEST_TEMPLATE.md]: @zachthompson Hmm, I guess - would it be worth sticking an `etc.` on the end there, to make it clear that those *sorts* of things should use the same format.

Comment (zachthompson) [inline .github/PULL_REQUEST_TEMPLATE.md]: Probably, yes.

Comment (GuiltyDolphin) [inline .github/PULL_REQUEST_TEMPLATE.md]: @zachthompson How's that?

**Comment by GuiltyDolphin** [comment]
@zachthompson @moollaza Shall we merge this and see how it goes? We can add more information later on if so desired.

**Comment by aanarres** [comment]
Is "Role" one of template, test, documentation, etc? If yes, I suggest

   - [ ] New Role (Template, Test, Documentation, etc.): **`New {Role Name} Role`**
   - [ ] Other: **`{{Role Name} Role}: {Short Description}`**

If not, what is it? :smile:

**Comment by GuiltyDolphin** [comment]
@aanarres GoodieRoles are Moo roles that can be loaded into Goodies; they have their own structures and are in themselves large components of the Instant Answer system.

Things like `template` and `documentation` are generally restricted to a specific non-IA region (the documentation does not directly change any IAs/Roles, nor do the templates).

I'd say a general rule is, if it is under `lib` or `share` it is going to be of the Instant Answer or Role format; otherwise it'll be under the Other format.

**Comment by aanarres** [comment]
Ah okay, thank you @GuiltyDolphin. Well, I wouldn't have known what to understand from roles, and I would have messed up. Now that I know what to look for, I found them in documentation under 'Helpers'.

Maybe:

- [ ] New Instant Answer
    - [ ] Cheat Sheets: **`New {Cheat Sheet Name} Cheat Sheet`**
    - [ ] Other: **`New {IA Name} Instant Answer`**
- [ ] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] New GoodieRole: **`New {Role Name} GoodieRole`**
- [ ] Other (GoodieRole, Template, Test, Documentation, etc.): **`{{Name} GoodieRole/Templates/Tests/Docs}: {Short Description}`**

Or your last version. It makes sense now, that I wrap my head around it. How about using GoodieRole instead of Role, and Name instead of "Role Name" in the last lines, where it stands for templates and tests and stuffs in addition to roles?

**Comment by GuiltyDolphin** [comment]
@aanarres They're actually called Roles - I just started calling them GoodieRoles at some point (and prefer the name :wink:); @zachthompson thoughts? Keep Role for the checkmark, or switch to GoodieRole?

**Comment by zachthompson** [comment]
@GuiltyDolphin Roles are duplicated.  I don't know whether they really deserve their own section.  Don't get created or changed very often.  `Other` would probably suffice.

**Comment by GuiltyDolphin** [comment]
@zachthompson There are at least 4 open Role PRs :wink: but okay - I'll move it to other.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+your Pull Request.
+
+- [ ] New Instant Answer
+    - [ ] Cheat Sheets: **`New {Cheat Sheet Name} Cheat Sheet`**
+    - [ ] Other: **`New {IA Name} Instant Answer`**
+- [ ] Improvement
+    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
+    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
+- [ ] Non–Instant Answer
+    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{{Role Name} GoodieRole/Templates/Tests/Docs}: {Short Description}`**
```
Comment (zachthompson) [inline .github/PULL_REQUEST_TEMPLATE.md]: Extra `{`.  Let's think of something else besides `Role Name`.  Not all of these are roles.  Potentially we could just remove that part:

* GoodieRole: New bleh blah role
* Templates: New template for X
* Tests: Switched to Test::Everything globally
* Docs: Overhauled the entire doc ecosystem

Comment (GuiltyDolphin) [inline .github/PULL_REQUEST_TEMPLATE.md]: @zachthompson Indeed - the `{Role Name}` was supposed to be grouped with `GoodieRole`; but seems to have wondered a bit.

Removal seems like it should work.

**Comment by zachthompson** [comment]
I dig it, thank you.

**Pull request merged**